### PR TITLE
feat: backport avatar component functionality from vue version

### PR DESCRIPTION
**Pull request by beneditomauro**
| - | - |
| --- | --- |
| Issue | |
| Description | Add image fallack functionality onError image loading |
| Screenshot | Captura de ecrã 2021-05-25, às 08 48 12|

### CODE REVIEW

**Review by NFSS10 — CHANGES_REQUESTED** [review]
The build is failling.
It looks like there is a problem with the linter, maybe the script `pretty` is not working properly, take a look a that

File: react/components/atoms/avatar/avatar.js [hunk 1]
```diff
@@ -60,9 +69,13 @@ export class Avatar extends mix(PureComponent).with(IdentifiableMixin) {
                 hitSlop={this.props.hitSlop}
             >
                 <Image
-                    source={this.props.image}
+                    source={this.state.imageSrc}
                     style={styles.image}
                     resizeMode={this.props.resizeMode}
+                    onError={() => {
```
Comment (gcandal) [inline react/components/atoms/avatar/avatar.js:75]: Extract to `onError` function

File: react/components/atoms/avatar/avatar.js [hunk 2]
```diff
@@ -60,9 +69,13 @@ export class Avatar extends mix(PureComponent).with(IdentifiableMixin) {
                 hitSlop={this.props.hitSlop}
             >
                 <Image
-                    source={this.props.image}
+                    source={this.state.imageSrc}
                     style={styles.image}
                     resizeMode={this.props.resizeMode}
+                    onError={() => {
+                        this.setState({ imageSrc: require("./assets/avatar.png") });
+                        this.props.onError;
```
Comment (gcandal) [inline react/components/atoms/avatar/avatar.js:77]:
```suggestion
                        if (this.props.onError) this.props.onError();
```

**Review by gcandal — CHANGES_REQUESTED** [review]

File: react/components/atoms/avatar/avatar.js (hunk 1 above)
Comment (beneditomauro) [reply react/components/atoms/avatar/avatar.js:75]: I'm afraid I cannot access` this.state` in `defaultProps()` function

Unless I missed something, you asked to define this directly in `onError` prop

**Review by gcandal — APPROVED** [review]

**Review by joamag — APPROVED** [review]

**Pull request merged**